### PR TITLE
ddl: don't rely on `expression.Column.ColName`

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Clean the dependencies of `expression.Column`.

### What is changed and how it works?

Directly extract the columns. Don't build expression and use `expression.Column.ColName`.

### Check List 

Tests 

 - Existing Unit test

### CODE REVIEW

File: ddl/partition.go [hunk 1]
```diff
@@ -634,16 +635,49 @@ func checkPartitionKeysConstraint(sctx sessionctx.Context, partExpr string, idxC
...
+		}
+		cne.err = ErrBadField.GenWithStackByArgs(c.Name.Name.O, "expression")
+		return nil, false
+	}
+	return node, true
+}
+
+func extractPartitionColumns(partExpr string, tblInfo *model.TableInfo) ([]*model.ColumnInfo, error) {
+	partExpr = "select " + partExpr
+	stmts, _, err := parser.New().Parse(partExpr, "", "")
```
Comment (lamxTyler) [inline ddl/partition.go]: Should we handle warnings like `ParseSimpleExprWithTableInfo`?

Comment (winoros) [reply ddl/partition.go]: I think we don't need. The warning generated before is not handled either.

File: ddl/partition.go [hunk 2]
```diff
@@ -634,16 +635,49 @@ func checkPartitionKeysConstraint(sctx sessionctx.Context, partExpr string, idxC
 	return nil
 }
 
-func extractPartitionColumns(sctx sessionctx.Context, partExpr string, tblInfo *model.TableInfo) ([]*expression.Column, error) {
-	e, err := expression.ParseSimpleExprWithTableInfo(sctx, partExpr, tblInfo)
+type columnNameExtractor struct {
```
Comment (qw4990) [inline ddl/partition.go]: Please add some unit tests for this new structure.

Comment (winoros) [reply ddl/partition.go]: I think that the current tests covered it? It's a code change, not some new codes

**Comment by tiancaiamao** [comment]
LGTM

File: ddl/partition.go [hunk 3]
```diff
@@ -634,16 +635,49 @@ func checkPartitionKeysConstraint(sctx sessionctx.Context, partExpr string, idxC
...
+func extractPartitionColumns(partExpr string, tblInfo *model.TableInfo) ([]*model.ColumnInfo, error) {
+	partExpr = "select " + partExpr
+	stmts, _, err := parser.New().Parse(partExpr, "", "")
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, err
+	}
+	extractor := &columnNameExtractor{
+		tblInfo:          tblInfo,
+		extractedColumns: make([]*model.ColumnInfo, 0),
```
Comment (ngaut) [inline ddl/partition.go]: You can keep as nil

**Pull request merged**